### PR TITLE
Refactor runtest and improve output readability.

### PR DESCRIPTION
**Pull request by cassiersg**
As part of my work I needed to add new tests and ended up refactoring the `compiler/scripts/rutest` python script.

I also modified a bit the output format to be quiter about tests that pass (only kept a status line at the bottom with current progress and last test run).

Example output:
```
scripts/runtest --jobs="2" config/tests.config --report=report.log safety CCT x86-64-ATT x86-64-Intel x86-64-print x86-64-nolea arm-m4 arm-m4-print x86-64-stack-zero-loop x86-64-stack-zero-unrolled arm-m4-stack-zero-loop arm-m4-stack-zero-unrolled
Failed test ./scripts/check-x86-64 -ATT tests/success/x86-64/aes.jazz
--------------- Stdout ---------------

--------------- Stderr ---------------
+ dirname ./scripts/check-x86-64
+ ./scripts/../jasminc -g -o jasminAkFKxD/jasmin.s -ATT tests/success/x86-64/aes.jazz
"tests/success/x86-64/aes.jazz", line 50 (17-26)
warning: unused variable next_rkey.191
+ echo -ATT tests/success/x86-64/aes.jazz
+ grep -q fail
+ cc -c -o jasminAkFKxD/jasmin.o jasminAkFKxD/jasmin.s
jasminAkFKxD/jasmin.s: Assembler messages:
jasminAkFKxD/jasmin.s:130: Error: operand type mismatch for `vaeskeygenassist'
+ rm -r jasminAkFKxD


Failed test ./scripts/check-x86-64 -intel tests/success/x86-64/aes.jazz
--------------- Stdout ---------------

--------------- Stderr ---------------
+ dirname ./scripts/check-x86-64
+ ./scripts/../jasminc -g -o jasminmxWlZj/jasmin.s -intel tests/success/x86-64/aes.jazz
"tests/success/x86-64/aes.jazz", line 50 (17-26)
warning: unused variable next_rkey.191
+ echo -intel tests/success/x86-64/aes.jazz
+ grep -q fail
+ cc -c -o jasminmxWlZj/jasmin.o jasminmxWlZj/jasmin.s
jasminmxWlZj/jasmin.s: Assembler messages:
jasminmxWlZj/jasmin.s:130: Error: operand type mismatch for `vaeskeygenassist'
+ rm -r jasminmxWlZj

Tests:     453/2200 | Failed:    2 | [SUCCESS] ./scripts/parse-print-parse tests/success/x86-64/test_mem.jazz
```
Feel free to close if it's not something you'd like to have upstream (I can also make tweaks if needed).

### CODE REVIEW

**Review by vbgl — APPROVED** [review]
Thanks for your contribution. I do not understand the changes but the output looks nice.

Note that there is a `AUTHORS` file in which you may want to put your name. Waiting for your reply before merging.

**Comment by cassiersg** [comment]
@vbgl updated

**Review by vbgl — APPROVED** [review]

**Pull request merged**
